searcher_client/src/lib.rs

```diff
@@ -54,3 +66,84 @@ pub async fn create_grpc_channel(url: &str) -> BlockEngineConnectionResult<Chann
     }
     Ok(endpoint.connect().await?)
 }
+
+pub async fn send_bundle_with_confirmation(
+    transactions: &[VersionedTransaction],
+    rpc_client: &RpcClient,
+    searcher_client: &mut SearcherServiceClient<InterceptedService<Channel, ClientInterceptor>>,
+    bundle_results_subscription: &mut Streaming<BundleResult>,
+) -> Result<(), Box<dyn std::error::Error>> {
+    let bundle_signatures: Vec<Signature> =
+        transactions.iter().map(|tx| tx.signatures[0]).collect();
+
+    let result = send_bundle_no_wait(transactions, searcher_client).await?;
+
+    // grab uuid from block engine + wait for results
+    let uuid = result.into_inner().uuid;
+    info!("Bundle sent. UUID: {:?}", uuid);
+
+    info!("Waiting for 5 seconds to hear results...");
+    while let Ok(Some(Ok(results))) =
+        timeout(Duration::from_secs(5), bundle_results_subscription.next()).await
```

Comment:
should we decrement the timeout? if searcher running another process using same auth key this could perpetually loop